openedx/core/lib/block_cache/block_structure.py

```diff
@@ -0,0 +1,588 @@
+"""
+Module with family of classes for block structures.
+    BlockStructure - responsible for block existence and relations.
+    BlockStructureBlockData - responsible for block & transformer data.
+    BlockStructureXBlockData - responsible for xBlock data.
+
+The following internal data structures are implemented:
+    _BlockRelations - Data structure for a single block's relations.
+    _BlockData - Data structure for a single block's data.
+"""
+# pylint: disable=protected-access
+from collections import defaultdict
+from logging import getLogger
+
+from openedx.core.lib.graph_traversals import traverse_topologically, traverse_post_order
+
+from .exceptions import TransformerException
+
+
+logger = getLogger(__name__)  # pylint: disable=invalid-name
+
+
+# A dictionary key value for storing a transformer's version number.
+TRANSFORMER_VERSION_KEY = '_version'
+
+
+class _BlockRelations(object):
+    """
+    Data structure to encapsulate relationships for a single block.
+
+    Attributes:
+        parents - a list of usage keys for this block's parent blocks
+        children - a list of usage keys for this block's child blocks
+    """
+    def __init__(self):
+
+        # list [UsageKey]
```

Comment:
nit: These inline comments seem redundant with the docstring for this small function.